### PR TITLE
Implement current_client

### PR DESCRIPTION
**Pull request by andreacerulli**

### CODE REVIEW

File: index_rs/src/index_rs/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 use ic_cdk::api::call::{CallResult, RejectionCode};
+use ic_cdk::caller;
```
Comment (robin-kunzler) [inline index_rs/src/index_rs/src/lib.rs]: nit: can be removed, also `RejectionCode` can be removed.

File: index_rs/src/index_rs/tests/tests.sh [hunk 2]
```diff
@@ -17,6 +17,10 @@ call IndexCanister.register_client(record
           { client_canister_id = client_canister_id });
 assert _ != (0 : nat);
 
+// should successfully query the current client
+call IndexCanister.current_client();
+assert _ != (null : opt null);
```
Comment (robin-kunzler) [inline index_rs/src/index_rs/tests/tests.sh]: nit: we could be a bit more specific as follows: 
```suggestion
assert _ == opt record { client_canister_id = client_canister_id };
```

**Review by robin-kunzler — APPROVED** [review]
looks good, find two minor comments below.

File: index_rs/src/index_rs/src/lib.rs (hunk 1 above)
Comment (andreacerulli) [reply index_rs/src/index_rs/src/lib.rs]: done

File: index_rs/src/index_rs/tests/tests.sh (hunk 2 above)
Comment (andreacerulli) [reply index_rs/src/index_rs/tests/tests.sh]: done

**Review by fspreiss — APPROVED** [review]

**Pull request merged**